cromwell_tools/_cromwell_auth.py

```diff
@@ -0,0 +1,91 @@
+import json
+import requests
+import requests.auth
+from oauth2client.service_account import ServiceAccountCredentials
+from ._cromwell_api import CromwellAPI
+
+
+class AuthenticationError(Exception):
+    pass
+
+
+class CromwellAuth:
+
+    def __init__(self, url, header, auth):
+        """Authentication Helper for a Cromwell Server
+
+        :param dict header: authorization header
+        """
+        if not header and not auth:
+            raise ValueError("either header or auth must be passed")
+
+        if header is not None:
+            if not isinstance(header, dict):
+                raise TypeError('if passed, header must be a dict')
+            if "Authorization" not in header.keys():
+                raise TypeError('the header must have an "Authorization" key')
+        self.header = header
+
+        # what is the type and what checks can be run?
+        self.auth = auth
+
+        if isinstance(url, str) and 'http' in url:
+            self.url = url
+        else:
+            raise ValueError("url must be an str that points to an http endpoint.")
+
+        # todo might need to change this re: cyclical import
+        # todo the health API endpoint wants an instance of this class, not just the auth object.
```

Comment:
If there's no clear way around the cyclical import, you could raise this error in the CromwellAPI methods

Reply:
Also I would suggest moving this check into a separate function so that it's easier to test

Reply:
You could consider doing that with the field validation as well, so you can test that more directly

Reply:
Makes sense. The function can't be called as it is, but i'll move some kind of check to its own class, and will do it with the other field validations as well.